cnxepub/html_parsers.py

```diff
@@ -16,11 +18,13 @@
     }
 
 
-def _squash_to_text(elm):
+def _squash_to_text(elm, remove_namespaces=False):
     value = [elm.text or '']
     for child in elm.getchildren():
-        value.append(etree.tostring(child).decode('utf-8'))
+        value.append(etree.tostring(child).decode('utf-8').strip())
         value.append(child.tail or '')
+    if remove_namespaces:
+        value = [re.sub(' xmlns:?[^=]*="[^"]*"', '', v) for v in value]
```

Comment:
Ah, you're right it gets them all - I wasn't seeing recursion, but etree.tostring() does the entire subtree.